yt/data_objects/data_containers.py

```diff
@@ -643,7 +653,7 @@ def save_as_dataset(self, filename=None, fields=None):
 
         extra_attrs = dict([(arg, getattr(self, arg, None))
                             for arg in self._con_args + self._tds_attrs])
-        extra_attrs["con_args"] = self._con_args
+        extra_attrs["con_args"] = repr(self._con_args)
```

Comment:
I'm not sure I understand why we need to do this -- `_con_args` is a tuple of strings, so it should serialize correctly in h5py without this.  Now we're turning the tuple into a string that represents the tuple, yes?

Reply:
h5py serializes a tuple of strings as a string array, this was leading to a case where I wasn't able to round-trip `con_args` anymore. Using `repr` and `eval` seemed like the most straightforward way to take care of that.